### PR TITLE
Use host id for authentication

### PR DESCRIPTION
**Pull request by alphadose**

### CODE REVIEW

**Review by abhishek9686 — CHANGES_REQUESTED** [review]
Authenticate should only be host basis
-> node is not a required parameter anymore to this func
-> should use different endpoint altogether for host authentication

**Comment by theguy951357** [comment]
will wait on changes before testing

File: functions/auth.go [hunk 1]
```diff
@@ -17,7 +17,7 @@ import (
 func Authenticate(node *config.Node, host *config.Config) (string, error) {
 	data := models.AuthParams{
 		MacAddress: host.MacAddress.String(),
-		ID:         node.ID.String(),
+		ID:         host.ID.String(),
```
Comment (mattkasun) [inline functions/auth.go:20]: node can be removed from func params

**Review by mattkasun — CHANGES_REQUESTED** [review]

File: functions/auth.go (hunk 1 above)
Comment (alphadose) [reply functions/auth.go:19]: done

**Comment by theguy951357** [comment]
go build is giving me this
 # github.com/gravitl/netclient/functions
functions/join.go:285:28: invalid operation: cannot indirect defaultInterface (variable of type string)
functions/join.go:286:40: invalid operation: cannot indirect defaultInterface (variable of type string)

**Review by 0xdcarns — APPROVED** [review]

File: functions/auth.go [hunk 2]
```diff
@@ -14,19 +13,15 @@ import (
...
 	}
-	server := config.GetServer(node.Server)
-	if server == nil {
-		return "", errors.New("nil server")
-	}
 	endpoint := httpclient.Endpoint{
-		URL:    "https://" + server.API,
-		Route:  "/api/nodes/adm/" + node.Network + "/authenticate",
+		URL:    "https://" + url,
+		Route:  "/api/hosts/adm/" + network + "/authenticate",
```
Comment (abhishek9686) [inline functions/auth.go:24]: host auth shouldn't be associated with network

**Review by abhishek9686 — CHANGES_REQUESTED** [review]

**Comment by theguy951357** [comment]
will wait on changes

File: functions/auth.go (hunk 2 above)
Comment (alphadose) [reply functions/auth.go:24]: but network is required for creating JWT
```
tokenString, err := logic.CreateJWT(authRequest.ID, authRequest.MacAddress, mux.Vars(request)["network"])
```

**Review by theguy951357 — APPROVED** [review]
As is, this PR passes. testing notes are in the kitemaker ticket. I can retest after changes if needed.

File: functions/auth.go (hunk 2 above)
Comment (0xdcarns) [reply functions/auth.go:24]: we should remove need for network specification for host auth

Comment (alphadose) [reply functions/auth.go]: done

**Review by abhishek9686 — APPROVED** [review]

**Comment by theguy951357** [comment]
retesting

**Review by theguy951357 — APPROVED** [review]
made nodes and server off of branches
joined network
netclient list shows the right information
pass

**Pull request merged**